airgun/views/host_new.py

```diff
@@ -264,12 +264,114 @@ class traces(Tab):
         enable_traces = OUIAButton('OUIA-Generated-Button-primary-1')
 
     @View.nested
-    class ansible(Tab):
+    class insights(Tab):
         pass
 
     @View.nested
-    class insights(Tab):
-        pass
+    class ansible(Tab):
+        """View comprising the subtabs under the Ansible Tab"""
+
+        ROOT = './/div'
+
+        @View.nested
+        class roles(Tab):
+            TAB_NAME = 'Roles'
+            ROOT = './/div[@class="ansible-host-detail"]'
+
+            assignedRoles = Text('.//a[contains(@href, "roles/all")]')
+            edit = Button(locator='.//button[@aria-label="edit ansible roles"]')
+            table = Table(
+                locator='.//table[contains(@class, "pf-c-table")]',
+                column_widgets={'Name': Text('.//a')},
+            )
+            pagination = Pagination()
+
+        @View.nested
+        class variables(Tab):
+            TAB_NAME = 'Variables'
+            ROOT = './/div[@class="ansible-host-detail"]'
+            table = Table(
+                locator='.//table[contains(@class, "pf-c-table")]',
+                column_widgets={
+                    'Name': Text('.//a'),
+                    'Ansible role': Text('./span'),
+                    'Type': Text('./span'),
+                    # the next field can also be a form group
+                    'Value': Text('./span'),
+                    'Source attribute': Text('./span'),
+                    # The next 2 buttons are hidden by default, but appear in this order
+                    5: Button(locator='.//button[@aria-label="Cancel editing override button"]'),
+                    6: Button(locator='.//button[@aria-label="Submit override button"]'),
+                    # Clicking this button hides it, and displays the previous 2
+                    7: Button(locator='.//button[@aria-label="Edit override button"]'),
+                },
+            )
+            pagination = Pagination()
+
+        @View.nested
+        class inventory(Tab):
+            TAB_NAME = 'Inventory'
+            ROOT = './/div[@class="ansible-host-detail"]'
+
+        @View.nested
+        class jobs(Tab):
+            TAB_NAME = 'Jobs'
+            ROOT = './/div[@class="ansible-host-detail"]'
+
+            @property
+            def is_displayed(self):
+                return (
+                    self.schedule.is_displayed
+                    or self.jobs.is_displayed
+                    or self.previous.is_displayed
+                )
+
+            class schedule(Tab):
```

Comment:
I think you are missing the `@View.nested` decorator on these classes :thinking:

Reply:
Seems like it, yep